syntax = "proto3";

service NumStore {
        rpc SetNum(SetNumRequest) returns (SetNumResp);

        rpc Fact(FactRequest) returns (FactResp);
}

message SetNumRequest {
        string key = 1;
        int32 value = 2;
}

message SetNumResp {
        int32 total = 1;
}

message FactRequest {
        string key = 3;
}

message FactResp {
        int32 value = 3;
        bool hit = 4;
        string error = 5;
}
